### PR TITLE
[Non-urgent] Disease correlation cleanup

### PR DESCRIPTION
**Pull request by jaclynbeck-sage**
### Cleanup of disease correlation transform/tests

While working on the `normalize_null_values` function (separate PR), I ended up needing to edit the disease_correlation tests but was having a lot of difficulty figuring out  the test file because it had a lot of copy/pasted data frames. I couldn't keep track of what needed to be edited or what tests needed to be added or removed in order to update the transform.

I made some major edits to the test file and some minor edits/cleanup to the transform itself in preparation for normalizing null values, but it bloated the `normalize_null_values` PR, so I split this cleanup off into its own PR.

**This PR does not depend on any of the other PRs, and no PRs depend on it being merged.**

#### Transform / Utils
- Removed unnecessary column rename ("model" -> "name")
- Removed the now-defunct function `input_validation_model_info`, which was probably written for an earlier version of the model_info source file. It checks for consistency between rows with the same model name, but now there is only one row per model so this check is unnecessary.
    - Also removed the associated tests from `utils.py` and `test_disease_correlation.py`
- Removed all `.fillna("")` and replaced NaN values with None in the numeric columns (as would be done with `normalize_null_values`
    - This allowed me to simplify some of the code due to not having to convert strings back to floats or check for NaN or empty strings
- Ensured that data with missing values in the grouping columns or the module column is dropped. 
    - Previously, data was dropped by `.groupby()` if it was missing from the grouping columns, but it wasn't dropped when module was missing. 
    - This is a defensive change and doesn't affect the current output. Realistically, when we add column value validation to this transform we should probably enforce that none of these columns have missing values.

#### Tests
- Removed some redundant/unncessary tests from `test_disease_correlation.py`:
    - "Test case 1" in the original `TestTransformDiseaseCorrelation` class was testing perfect data, which was already being tested by the asset test
    - "Test case 2" in the original `TestTransformDiseaseCorrelation` class tested when there were two rows in allele_info that were exact duplicates of each other, which didn't seem useful to test since we control that source file
    - "Error test case 3" in the original `TestTransformDiseaseCorrelation` class tested `input_validation_model_info`, so it was removed.
- Changed "Error test case 1" (missing dataset) and "Error test case 2" (duplicate module name) over to using asset files to shorten code
- New expected failure test for datasets missing a required column
- New passing test with assets: rows missing values in grouping columns or the module column should get removed
- New passing test with assets: rows missing correlation, p-value, or both are handled gracefully
- New passing test in the `TestProcessGroup` class: tests that data with missing correlation and/or p-value is handled gracefully by `process_group`.
    - This is slightly redundant with the asset test, however the asset test is testing end-to-end and looking at the final output, while this function is a unit test for the function itself and only looks at the single dictionary output by the function.

### CODE REVIEW

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 1]
```diff
@@ -186,25 +184,19 @@ def process_group(
...
     for _, row in group.iterrows():
-        if extract_module_name(row["module"]) in output:
+        module_name = extract_module_name(row["module"])
+        if module_name in output:
             raise ValueError(
-                f"Module {extract_module_name(row['module'])} already exists for {output['name']}"
+                f"Module {module_name} already exists for {output['name']}"
             )
 
-        module_dict = {
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/disease_correlation.py:194]: Removed the .fillna("") at the beginning of the transform and replaced NaN with None, so these values are now already floats, and missing values are already None.

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 2]
```diff
@@ -253,30 +245,35 @@ def transform_disease_correlation(
...
-    disease_correlation_df = datasets["disease_correlation_results"].fillna("")
-    model_info_df = datasets["model_info"].fillna("")
-    allele_info_df = datasets["allele_info"].fillna("")
-    human_transgene_allele_map_df = datasets["human_transgene_allele_map"].fillna("")
-
-    # Validate model info
-    input_validation_model_info(model_info_df)
+    disease_correlation_df = datasets["disease_correlation_results"].replace(
+        {"correlation": {np.nan: None}, "adjusted_p_value": {np.nan: None}}
+    )
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/transform/disease_correlation.py]: This will be replaced with `disease_correlation_df = normalize_null_values(datasets["disease_correlation_results"])` when that PR is ready.

File: src/agoradatatools/etl/utils.py [hunk 3]
```diff
@@ -365,34 +365,6 @@ def convert_numpy_types(obj: Any) -> Any:
     return obj
 
 
-def input_validation_model_info(df: pd.DataFrame) -> None:
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/etl/utils.py:608]: This function might be an artifact of an earlier version/format of model_info and is no longer needed. There should only be one row per model name in the source data now.

File: tests/transform/test_disease_correlation.py [hunk 4]
```diff
@@ -73,390 +105,92 @@ def test_disease_correlation_transform_assets_should_pass(
         # Compare actual output with expected output
         assert output_data == expected_data
 
-
-class TestTransformDiseaseCorrelation:
```
Comment (jaclynbeck-sage) [inline tests/transform/test_disease_correlation.py:77]: All of these tests were moved to using asset files instead, or removed:

1. Pass test case 1: basically testing the same thing as the first asset test above, so it was removed
2. Pass test case 2: removed, since we control the source allele info file there shouldn't be any duplicated rows. I didn't see a need to make a test for it.
3. Error test case 1: new test function that reads normal asset files and deletes a data set one at a time
4. Error test case 2: new test function with an asset file that has a duplicated module
5. Error test case 3: removed, we no longer test for inconsistent model_info since that's not how the file is structured anymore
6. Column error test case: new test function that reads normal asset files and deletes a required column from each data set

File: tests/test_utils.py [hunk 5]
```diff
@@ -729,110 +729,6 @@ def test_remove_duplicates_preserves_order(self):
         assert utils.remove_duplicates_keep_order(input_list) == ["a", "b", "c", "d"]
 
 
-class TestInputValidationModelInfo:
```
Comment (jaclynbeck-sage) [inline tests/test_utils.py:1166]: Removed since the function was removed.

File: tests/transform/test_disease_correlation.py [hunk 6]
```diff
@@ -541,19 +286,19 @@ def test_process_group_with_valid_data(self):
             [
                 {
                     "module": "IFGyellow",
-                    "correlation": "0.5",
-                    "adjusted_p_value": "0.01",
+                    "correlation": 0.5,
+                    "adjusted_p_value": 0.01,
```
Comment (jaclynbeck-sage) [inline tests/transform/test_disease_correlation.py:295]: With the removal of the .fillna(""), these values are now floats, not strings.

File: tests/transform/test_disease_correlation.py [hunk 7]
```diff
@@ -73,390 +114,94 @@ def test_disease_correlation_transform_assets_should_pass(
...
-        column_error_test_data,
-        ids=column_error_test_ids,
+        "dataset, missing_column",
+        [
+            ("disease_correlation_results", "cluster"),
+            ("model_info", "model"),
+            ("allele_info", "model"),
+            ("human_transgene_allele_map", "mgi_allele_id"),
+        ],
+        ids=[],
```
[the remark this answers is not in the document]
Comment (jaclynbeck-sage) [reply tests/transform/test_disease_correlation.py]: Fixed in next commit.

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 8]
```diff
@@ -253,30 +245,35 @@ def transform_disease_correlation(
...
 
     # Map mouse gene names to human gene symbols
     allele_info_mapped = map_genes_to_human_symbols(
         allele_info_df, human_transgene_allele_map_df
     )
 
-    # Need convert 'matched_controls' from comma-separated strings to lists
+    # Need to convert 'matched_controls' from comma-separated strings to lists
     model_info_df["matched_controls"] = model_info_df["matched_controls"].apply(
         lambda x: delim_string_to_list(x, delim=",")
```
[the remark this answers is not in the document]
Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/disease_correlation.py:262]: Fixed in next commit

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 9]
```diff
@@ -28,12 +28,12 @@
...
+        "model",
         "matched_controls",
         "model_type",
     ],
     "allele_info": [
-        "name",
+        "model",
         "gene",
         "mgi_allele_id",
     ],
```
[the remark this answers is not in the document]
Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/disease_correlation.py:39]: Fixed in next commit

File: tests/transform/test_disease_correlation.py [hunk 10]
```diff
@@ -644,6 +394,74 @@ def test_process_group_with_list_matched_controls(self):
         # Should take first element from the list
         assert result["matched_control"] == "C57BL6J"
 
+    def test_process_group_with_none_values(self) -> None:
+        """
+        Test that process_group correctly handles None values in the disease correlation DataFrame.
```
Comment (linglp) [inline tests/transform/test_disease_correlation.py]: Nit: I think the test function name and docstring should mention about zero since this is an important part of the test. It should be something like: Test that process_group correctly handles None and zero values in the disease correlation DataFrame.

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 11]
```diff
@@ -161,15 +161,13 @@ def process_group(
...
+    # Get the first list element of matched_controls, default to empty string if not present
+    mc = model_info.get("matched_controls", [])
+    matched_control = next(iter(mc), "")
 
     # Ensure modified_genes is always a list
-    raw_modified_genes = allele_info.get("gene", "")
-    if raw_modified_genes == "":
-        modified_genes = []
-    elif not isinstance(raw_modified_genes, list):
+    raw_modified_genes = allele_info.get("gene", [])
```
Comment (linglp) [inline src/agoradatatools/etl/transform/disease_correlation.py:169]: hmm this is interesting, and I think we should probably create another ticket to fix it if we want to. Here, it seems like `allele_info.get("gene")` can either be a string or a list from upstream, but we always want it to be a list in the downstream.

If gene is always expected to be a list, create_lookup should normalize single values to lists before returning, so callers never need to defensive-check isinstance(..., list).

But this can be risky because I am not sure how that would affect other places calling `create_lookup`. 

This is also not urgent since here is the only place that you have to be defensive.

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 12]
```diff
@@ -253,30 +245,35 @@ def transform_disease_correlation(
...
+    model_info_lookup = create_lookup(model_info_df, group_by_col="model")
 
-    model_allele_lookup = create_lookup(df=allele_info_mapped, group_by_col="name")
+    model_allele_lookup = create_lookup(df=allele_info_mapped, group_by_col="model")
 
     # Group by all static fields
     output = []
     group_cols = ["mouse_model", "cluster", "age", "sex"]
+
+    # Drop any rows with missing values in the grouping columns or the module column
```
Comment (linglp) [inline src/agoradatatools/etl/transform/disease_correlation.py:272]: Nit: I think here the comment should reflect the "why" instead of "what". Why do we want to drop rows with `NaN` in module?

File: src/agoradatatools/etl/transform/disease_correlation.py [hunk 13]
```diff
@@ -161,15 +161,13 @@ def process_group(
     Returns:
         Dict[str, Any]: A dictionary containing the processed group data
     """
-    # If matched_controls is a list, get the first element
-    mc = model_info.get("matched_controls", "")
-    matched_control = next(iter(mc), "") if isinstance(mc, list) else mc
+    # Get the first list element of matched_controls, default to empty string if not present
+    mc = model_info.get("matched_controls", [])
```
Comment (linglp) [inline src/agoradatatools/etl/transform/disease_correlation.py:165]: It seems like if there's multiple `matched_control`, we will only take the first element from `matched_control`? I am curious why is that the case?

**Review by linglp — APPROVED** [review]
LGTM! Thanks for the cleanup. I just have some minor comments :)

File: src/agoradatatools/etl/transform/disease_correlation.py (hunk 13 above)
Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/disease_correlation.py:165]: It's to deal with UCI's 2-in-one-study setup. Basically they have two related models they want to display on one page, so the group will have two matched controls like ["C57B6J", "5XFAD"]. The first one is in the list is always the "base"/unaltered mouse, so that's the one we mark as the matched control. We make the source file that gives that info, so we've just made sure the correct one is always first in the list. Not ideal, but it's what we could do in the time we had. :( Hopefully we can revisit this and make something cleaner at some point.

File: src/agoradatatools/etl/transform/disease_correlation.py (hunk 11 above)
Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/disease_correlation.py:169]: Interesting catch. I've created [this ticket](https://sagebionetworks.jira.com/browse/MG-910) to track!

File: src/agoradatatools/etl/transform/disease_correlation.py (hunk 12 above)
Comment (jaclynbeck-sage) [reply src/agoradatatools/etl/transform/disease_correlation.py:272]: After looking into this, I think we should use column rules to enforce that these values aren't NA in the first place. I've added them to [this WIP ticket](https://sagebionetworks.jira.com/browse/MG-907) to track this. So I am going to leave this comment as-is with the intent that this drop code should actually be removed later and replaced with validation.

File: tests/transform/test_disease_correlation.py (hunk 10 above)
Comment (jaclynbeck-sage) [reply tests/transform/test_disease_correlation.py]: Fixed in next commit.
